tests/SecurityGroup.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import json
+
+from Node import BlockType
+from Node import Node
+from Node import ReturnType
+from testUtils import Utils
+
+# pylint: disable=too-many-public-methods
+class SecurityGroup(object):
+
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=too-many-arguments
+    def __init__(self, nonParticipants, contractAccount, defaultNode=None, minAddRemEntriesToPublish=100):
+        self.participants = []
+        self.contractAccount = contractAccount
+        assert len(nonParticipants) > 0
+        self.nonParticipants = copy.deepcopy(nonParticipants)
+        if Utils.Debug: Utils.Print("Creating SecurityGroup with the following nonParticipants: []".format(SecurityGroup.createAction(self.nonParticipants)))
+        self.defaultNode = defaultNode if defaultNode else nonParticipants[0]
+        self.publishProcessNum = minAddRemEntriesToPublish
+        Utils.Print("Publish contract")
+        self.contractTrans = self.defaultNode.publishContract(self.contractAccount, "unittests/test-contracts/security_group_test/", "eosio.secgrp.wasm", "eosio.secgrp.abi", waitForTransBlock=True)
+        self.publishTrans = None
+
+    @staticmethod
+    def __lessThan(lhsNode, rhsNode):
+        assert lhsNode != rhsNode
+        return lhsNode.getParticipant() < rhsNode.getParticipant()
+
+    def __consistentStorage(self):
+        intersect = list(set([x.getParticipant() for x in self.participants]) & set([x.getParticipant() for x in self.nonParticipants]))
+        if len(intersect) != 0:
+            Utils.exitError("Values: {}, are in the nonParticipants list: {}, and the participants list: {}".format(", ".join(intersect), SecurityGroup.createAction(self.nonParticipants), SecurityGroup.createAction(self.participants)))
+
+    def __addParticipants(self, nodes):
+        if Utils.Debug: Utils.Print("Moving the following: {}, from the nonParticipants list: {}, to the participants list: {}".format(SecurityGroup.createAction(nodes), SecurityGroup.createAction(self.nonParticipants), SecurityGroup.createAction(self.participants)))
+        for node in nodes:
+            if node not in self.nonParticipants:
+                for nonParticipant in self.nonParticipants:
+                    match = "" if node == nonParticipant else " not"
```

Comment:
what is this loop for?
if `if node not in self.nonParticipants` was triggered how can you have a match here?

Reply:
Sorry, debug code, removed